onchaind/onchaind.c

```diff
@@ -152,6 +152,10 @@ static void trim_maximum_feerate(struct amount_sat funding,
 	size_t weight;
 	struct amount_sat fee = funding;
 
+	/* FIXME: This doesn't work for elements? */
+	if (chainparams->is_elements)
+		return;
```

Comment:
Good question, I don't think it should be too much of an issue, given that we currently only have one main-asset input, which also makes the below `if (!amount_asset_is_main(&amt))` always false, but better keep it in in case we eventually add multi-input and multi-asset channels.

Was there a concrete error that was triggered on elements, or is this more of a precaution?

Reply:
Yes, we had a -1ULL satoshis value. Then added this test and this function caused an onchaind failure (presumably by giving the wrong result).